mesonbuild/coredata.py

```diff
@@ -960,11 +960,16 @@ def __getitem__(self, key: OptionKey) -> options.UserOption:
                 opt.set_value(override_value)
         return opt
 
-    def get_value(self, key: T.Union[str, OptionKey]):
+    def get_value(self, key: T.Union[str, OptionKey]) -> ElementaryOptionValues:
         if isinstance(key, str):
             key = OptionKey(key)
         return self[key].value
 
+    def get_value_safe(self, key: T.Union[OptionKey, str], type_: T.Type[ElementaryOptionTypes]) -> ElementaryOptionTypes:
```

Comment:
Isn't it possible to tell that the returned type is given by `type_` ?

Reply:
That is exactly what's happening here, however the unsafe get inside could potentially return anything, so we need to do the assert to assure the type checker that it's getting what we told it it would get.